### ISSUE TITLE
LR Monitor does not work with SWA

### ISSUE THREAD

**Issue by aprbw** [PyTorchLightning/pytorch-lightning#9453]
## 🐛 Bug

LR Monitor does not work with SWA.

LR Monitor: [LearningRateMonitor](https://pytorch-lightning.readthedocs.io/en/latest/extensions/generated/pytorch_lightning.callbacks.LearningRateMonitor.html)

SWA: [Stochastic Weight Averaging](https://pytorch-lightning.readthedocs.io/en/latest/advanced/training_tricks.html#stochastic-weight-averaging)

### To Reproduce

https://colab.research.google.com/drive/1sksfJGj5d-_KJGVz3tM_ccOMnk9TPkys?usp=sharing

These lines in particular:
`lrmcb = pl.callbacks.LearningRateMonitor(logging_interval='step')`

`        stochastic_weight_avg=True,
        callbacks = [lrmcb]`

### Expected behavior

Not have this warning:

`/usr/local/lib/python3.7/dist-packages/pytorch_lightning/callbacks/lr_monitor.py:116: RuntimeWarning: You are using 'LearningRateMonitor' callback with models that have no learning rate schedulers. Please see documentation for 'configure_optimizers' method.
  RuntimeWarning,`

Because SWA is using a scheduler according to this: https://pytorch.org/blog/pytorch-1.6-now-includes-stochastic-weight-averaging/#how-to-use-swa-in-pytorch

And not have an empty list for the list of LR when inspecting the LearningRateMonitor callback instance:
`for v,k in vars(lrmcb).items():
    print(v,k)`
OUTPUT:
`logging_interval None
log_momentum False
lrs {'lr-SGD': []}
lr_sch_names []
log <bound method LightningModule.log of BoringModel(
  (layer): Linear(in_features=32, out_features=2, bias=True)
)>
log_dict <bound method LightningModule.log_dict of BoringModel(
  (layer): Linear(in_features=32, out_features=2, bias=True)
)>
last_momentum_values {}`

The line showing LR is an empty list:
`lrs {'lr-SGD': []}`

### Environment

* CUDA:
	- GPU:
	- available:         False
	- version:           10.2
* Packages:
	- numpy:             1.19.5
	- pyTorch_debug:     False
	- pyTorch_version:   1.9.0+cu102
	- pytorch-lightning: 1.4.6
	- tqdm:              4.62.0
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         x86_64
	- python:            3.7.11
	- version:           #1 SMP Sat Jun 5 09:50:34 PDT 2021

### Additional context

cc @tchaton @rohitgr7 @akihironitta

**Comment by rongcuid** [comment]
Same. The scheduler replaces any old scheduler, but LRMonitor does not seem to get updated, and instead just logs the last step's learning rate.

**Comment by aprbw** [comment]
Is still not working with the latest version pytorch-lightning: 1.5.0

**Comment by tchaton** [comment]
Dear @rongcuid @aprbw,

Would you be interested in contributing a fix?

Best,
T.C

**Comment by aprbw** [comment]
Yes, but I'm very not smart. I don't even know where to start.

**Comment by tchaton** [comment]
Hey @aprbw,

I believe the Learning Rate Monitor should check if the current schedulers are still used by the trainer and get the source of truth from there.

**Comment by aprbw** [comment]
Okay so, I tried it again with version 1.5.5

There are no more warnings, so that's good.

`lr-SGD` is also not empty anymore. That's also good.

However, the number stays constant at 0.1. To the best of my understanding, that's not the expected behavior as SWA scheduler is cyclical.

**Comment by tchaton** [comment]
Hey @aprbw,

Thanks for the update. We won't be able to prioritize this bug for the end of the quarter, I will keep you updated for the next one.

**Comment by rohitgr7** [comment]
in v1.5 release, we enabled logging learning rates even when there are no lr_schedulers returned from `configure_optimizers`. So this is working now. Tested with the attached script.

Feel free to reopen if it comes up again :)

**Comment by rohitgr7** [comment]
investigated up a bit into the issue. Looks like you need to set `swa_lrs` too.

else Lightning will pass in the original lr specified inside optimizer:
https://github.com/PyTorchLightning/pytorch-lightning/blob/3e0569fccc9808b1017ace63951485e2f76563eb/pytorch_lightning/callbacks/stochastic_weight_avg.py#L170-L171

and if you don't pass in the swa_lrs, in the following equation, lr and swa_lr will be the same thus `alpha` won't have any effect here:
https://github.com/pytorch/pytorch/blob/9267fd8d7395074001ad7cf2a8f28082dbff6b0b/torch/optim/swa_utils.py#L274

@tchaton do you think, we should make `swa_lrs` as required? since it's a required param in PyTorch [SWALR](https://github.com/pytorch/pytorch/blob/9267fd8d7395074001ad7cf2a8f28082dbff6b0b/torch/optim/swa_utils.py#L220), else we might see a similar issue in the future and SWA won't have any effect here in such a case.
